specs/customer_spec.rb

```diff
@@ -1,41 +1,71 @@
 require 'minitest/autorun'
 require 'minitest/reporters'
 require 'minitest/skip_dsl'
+require_relative '../lib/customer'
 
-# TODO: uncomment the next line once you start wave 3
-# require_relative '../lib/customer'
+Minitest::Reporters.use!
 
-xdescribe "Customer" do
+describe "Customer" do
   describe "#initialize" do
     it "Takes an ID, email and address info" do
-      # TODO: Your test code here!
+      id = 542
+      email = ""
+      address = []
+
+      customer = Grocery::Customer.new(id, email, address)
+
+      customer.must_respond_to :id
+      customer.id.must_equal id
+      customer.id.must_be_kind_of Integer
+      customer.email.must_be_kind_of String
+      customer.address.must_be_kind_of Array
     end
   end
 
   describe "Customer.all" do
     it "Returns an array of all customers" do
-      # TODO: Your test code here!
-      # Useful checks might include:
-      #   - Customer.all returns an array
-      #   - Everything in the array is a Customer
-      #   - The number of orders is correct
-      #   - The ID, email address of the first and last
-      #       customer match what's in the CSV file
-      # Feel free to split this into multiple tests if needed
-    end
-  end
+      array_of_customers = Grocery::Customer.all
 
-  describe "Customer.find" do
-    it "Can find the first customer from the CSV" do
-      # TODO: Your test code here!
+      array_of_customers.must_be_kind_of Array
+      array_of_customers.length.must_equal 35
     end
 
-    it "Can find the last customer from the CSV" do
-      # TODO: Your test code here!
+    it "Returns accurate information about the first customer" do
+      array_of_customers = Grocery::Customer.all
+      first_customer_id = array_of_customers[0][0]
+      first_customer_state = array_of_customers[0][2][2]
+
+      first_customer_id.must_equal 1
+      first_customer_state.must_equal "LA"
     end
 
-    it "Raises an error for a customer that doesn't exist" do
-      # TODO: Your test code here!
+    it "Returns accurate information about the last customer" do
+      array_of_customers = Grocery::Customer.all
+      last_customer_id = array_of_customers[-1][0]
+      last_customer_email = array_of_customers[-1][1]
+
+      last_customer_id.must_equal 35
+      last_customer_email.must_equal "rogers_koelpin@oconnell.org"
     end
   end
 end
+
+describe "Customer.find" do
+  it "Can find the first customer from the CSV" do
+    first_customer_info = Grocery::Customer.find(1)
+
+    first_customer_info.must_equal Grocery::Customer.all[0]
```

Comment:
You shouldn't test `find` against `all` since one depends on the other (circular testing).

Instead you should

```ruby
first_customer = Grocery::Customer.find(1)
first_customer.id.must_equal 1
first_customer.email.must_equal .....
```